tests/unit/apps/test_auxiliary.py

```diff
@@ -160,6 +164,23 @@ def test_auxiliary_upgrade_plan_ussuri_to_victoria_change_channel(model):
             parallel=False,
             coro=model.upgrade_charm(app.name, "3.8/stable"),
         ),
+    ]
+    if app.config["enable-auto-restarts"].get("value") is False:
```

Comment:
not-blocking: I'll prefer to have separate test instead of logic like this. Since this logic is simple it could be fine, but there is no harm to duplicate tests function to make them more readable.

Reply:
👍🏼